portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java

```diff
@@ -1176,21 +1176,27 @@ else if ((videoThumbnail > 0) && (videoThumbnail <= 3)) {
 
 		// Send file
 
-		httpServletResponse.addHeader(
-			HttpHeaders.CACHE_CONTROL,
-			FileEntryHttpHeaderCustomizerUtil.getHttpHeaderValue(
-				fileEntry, HttpHeaders.CACHE_CONTROL,
-				HttpHeaders.CACHE_CONTROL_PRIVATE_VALUE));
+		boolean download = ParamUtil.getBoolean(httpServletRequest, "download");
+
+		if (download) {
+			httpServletResponse.addHeader(
+				HttpHeaders.CACHE_CONTROL,
+				HttpHeaders.CACHE_CONTROL_NO_CACHE_VALUE);
```

Comment:
We should still call the header customizers, but change the default values:  "no-cache" when downloading, and "private" when embedding. Otherwise, we'll be breaking a feature: https://issues.liferay.com/browse/LPS-146748.

Reply:
Correct me if I'm wrong, but the security ticket want to "break" the feature always if the document is downloaded.
It seems pointless to change `FileEntryHttpHeaderCustomizerUtil` if downloaded will always will return the same value

I'm reading the issue wrong?

Reply:
thanks for the knowledge :D